services/QuillLMS/app/services/skills_report/skill_responses_processor.rb

```diff
@@ -40,6 +40,15 @@ def no_data_result
     private def avg_time_spent = avg_of(:timespent)
     private def number_of_students = student_skill_results.length
 
+    private def last_practiced
+      return nil unless student_skill_results.present?
+
+      last_practiced_dates = student_skill_results.map(&:last_practiced).compact
+      return nil if last_practiced_dates.empty?
```

Comment:
I don't think you actually need this guard clause because in Ruby `[].max` is `nil` already.  So I think you can just return `last_practiced_dates&.max`.